presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java

```diff
@@ -79,27 +76,25 @@ public void removeAgent(int id)
     //TODO: Add method to generate ID
     public static class Agent
     {
-        private final URI uri;
-        private static final AtomicInteger count = new AtomicInteger(0);
-        private final int id;
-        private final boolean isCoordinator;
-        private final boolean isWorker;
+        private URI uri;
+        private UUID id;
+        private boolean isCoordinator;
+        private boolean isWorker;
 
-        private Agent(URI uri, boolean isCoordinator, boolean isWorker)
+        private Agent(URI uri, boolean isCoordinator, boolean isWorker, UUID id)
         {
             this.uri = requireNonNull(uri, "uri is null");
             this.isCoordinator = isCoordinator;
             this.isWorker = isWorker;
-
-            this.id = count.incrementAndGet();
+            this.id = requireNonNull(id, "null agent id");
```

Comment:
I put them in parameter order.

We can make that sort of change when we switch to a map-based NodeSet (that's the next thing I'm planning after finishing the switch to airlift.)